lib/JunkDetection/CXX/CompilationDatabase.cpp

```diff
@@ -4,10 +4,37 @@
 #include <llvm/ADT/SmallString.h>
 #include <llvm/Support/Path.h>
 
+#include <regex>
 #include <sstream>
 
 using namespace mull;
 
+/// Compilation databases such as one created by CMake can be a subject of the
+/// following:
+///
+///     Parameters use shell quoting and shell escaping of quotes, with ‘"’ and
+///     ‘\’ being the only special characters
+///     https://clang.llvm.org/docs/JSONCompilationDatabase.html#format
+///
+/// The compile DB uses shell escaping, and this needs to be undone because
+/// clang::ASTUnit::LoadFromCommandLine() doesn't execute commands in shell.
+///
+/// See "Problem with compiler defines and use of '\"'" (#678) for details.
+/// https://github.com/mull-project/mull/issues/678
+///
+/// Example:
+/// {
+///   "directory": "/src/builds/amd64-linux-mull",
+///   "command":
+///     "/usr/bin/clang++-9 -DQT_TESTCASE_BUILDDIR=\\\"/src/builds/amd64-linux-mull\\\" ...",
+///   "file": /src/tests/TestingUnitTests/ModuleTestSuite.cpp"
+/// }
+static std::string undoEscapingString(std::string &string) {
+  std::string unescaped(std::regex_replace(string, std::regex(R"(\\\\)"), "\\"));
+  unescaped = std::regex_replace(string, std::regex(R"(\\\")"), "\"");
```

Comment:
1. Did you mean to pass the `unescaped` variable here instead of the `string` parameter?
2. Why regexes?

Reply:
1) you are right, I have changed it. Interesting that it made Clang to parse this correctly without the first statement to work.
2) Please suggest a good alternative.